schema/pgsql/schema.sql

```diff
@@ -90,45 +90,66 @@ CREATE TABLE schedule (
     CONSTRAINT pk_schedule PRIMARY KEY (id)
 );
 
+CREATE TABLE rotation (
+    id bigserial,
+    schedule_id bigint NOT NULL REFERENCES schedule(id),
+    -- the lower the more important, starting at 0, avoids the need to re-index upon addition
+    priority integer NOT NULL,
+    name text NOT NULL,
+    mode rotation_type NOT NULL,
+    -- JSON with rotation-specific attributes
+    -- Needed exclusively by Web to simplify editing and visualisation
+    options text NOT NULL,
+
+    -- A date in the format 'YYYY-MM-DD' when the first handoff should happen.
+    -- It is a string as handoffs are restricted to happen only once per day
+    first_handoff date NOT NULL,
+
+    -- each schedule can only have one rotation with a given priority starting at given date
+    UNIQUE (schedule_id, priority, first_handoff),
+
+    CONSTRAINT pk_rotation PRIMARY KEY (id)
+);
+
 CREATE TABLE timeperiod (
     id bigserial,
-    owned_by_schedule_id bigint REFERENCES schedule(id), -- nullable for future standalone timeperiods
+    owned_by_rotation_id bigint REFERENCES rotation(id), -- nullable for future standalone timeperiods
 
     CONSTRAINT pk_timeperiod PRIMARY KEY (id)
 );
 
 CREATE TABLE timeperiod_entry (
     id bigserial,
     timeperiod_id bigint NOT NULL REFERENCES timeperiod(id),
+    rotation_member_id bigint REFERENCES rotation_member(id), -- nullable for future standalone timeperiods
     start_time bigint NOT NULL,
     end_time bigint NOT NULL,
     -- Is needed by icinga-notifications-web to prefilter entries, which matches until this time and should be ignored by the daemon.
     until_time bigint,
     timezone text NOT NULL, -- e.g. 'Europe/Berlin', relevant for evaluating rrule (DST changes differ between zones)
     rrule text, -- recurrence rule (RFC5545)
-    -- Contains the same frequency types as in the rrule string except the `QUARTERLY` one, which is only offered
-    -- by web that is represented as `FREQ=MONTHLY;INTERVAL=3` in a RRule string. So, this should be also ignored
-    -- by the daemon.
-    frequency frequency_type,
-    description text,
 
     CONSTRAINT pk_timeperiod_entry PRIMARY KEY (id)
 );
 
-CREATE TABLE schedule_member (
-    schedule_id bigint NOT NULL REFERENCES schedule(id),
-    timeperiod_id bigint NOT NULL REFERENCES timeperiod(id),
+CREATE TABLE rotation_member (
```

Comment:
Table `rotation_member` must be created before `timeperiod_entry.`
Same in upgrade file.

Reply:
fixed